### PR TITLE
Cost estimates

### PR DESCRIPTION
**Pull request by abhagupta**
PR for #11   . Added unit test cases. Shows something like this at the end 
```
Operation completed over 6 objects/2.1 KiB.                                      
CommandException: No URLs matched: gs://apk-bucket-2017-06-01-09-40-05_1477703143/../walmart-android-qa-debug.apk
CommandException: No URLs matched: gs://apk-bucket-2017-06-01-09-40-05_1477703143/../walmart-android-qa-debug-androidTest.apk
Combined test execution time: ~2 minutes

End time: 2017-06-01_09:47

Total price incurred for virtual device is ~ $6.00
Total price incurred for real device is ~ $30.00

```

Billable time will show now:

```Combined test execution time: ~0 minutes

End time: 2017-06-01_01:56

Estimated costs:  $1.00(virtual) $5.00(physical) 
Billable time:  1 min(s) 
Process finished with exit code 0
```

Now it will show

```
Billable time:  6.0 min(s) 

Estimated cost:  $0.12(virtual) $0.48(physical) 

Combined test execution time: ~2 minutes

End time: 2017-06-01_04:57

```

### CODE REVIEW

File: Flank/src/main/java/com/walmart/otto/reporter/PriceReporter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import java.util.Map;
+
+public class PriceReporter {
+
+    private static HashMap<String, Double> mapOfPrices = new HashMap<String, Double>();
+    private static HashMap<String, BigDecimal> estimates = new HashMap<String, BigDecimal>();
+
+
+    public static HashMap<String, BigDecimal>  getTotalPrice(List<Integer> times)  {
+        mapOfPrices.put("virtual", (double)1);
```
Comment (abhagupta) [inline Flank/src/main/java/com/walmart/otto/reporter/PriceReporter.java]: I did not know where to create the Hashmap. Ideally this should be created somewhere in a Util file . Any suggestion here?

Comment (renas) [inline Flank/src/main/java/com/walmart/otto/reporter/PriceReporter.java]: I don't see anything wrong with creating it in this new class

**Comment by bootstraponline** [comment]
It'd also be nice to output the billable time in `hh:ss` format. Something like:

```
Combined test execution time: ~2 minutes
Billable time: ~10 minutes
```

With per method sharding, there's a surprising difference due to the rounding logic. It's about ~18x for us.

**Comment by abhagupta** [comment]
How do we calculate billable time? Is it some information that gcloud provide us?

**Comment by bootstraponline** [comment]
According to the flank readme, seconds are rounded up to minutes.

> seconds are round up to minutes by Firebase test lab
> https://github.com/TestArmada/flank

> Charges are calculated on a per-minute basis
> https://firebase.google.com/docs/test-lab/overview#billing

If we have 10 tests that take 2 seconds each running on 10 VMs, the combined execution time is 20 seconds. The real life waiting time is 2 seconds since they run in parallel. The billable time is 10 minutes due to rounding. Hope that makes sense.

**Comment by bootstraponline** [comment]
```
Operation completed over 1 objects.
Combined test execution time: ~42 minutes

End time: 2017-06-01_05:26

Estimated costs:  $113.00(virtual) $565.00(real) Billable time:  113 min(s)
```

I think something is wrong with the math?

**Comment by abhagupta** [comment]
ok let me fix the test and do one small update in display. then i will look at the math

**Comment by bootstraponline** [comment]
I think the issue is minutes are being confused with hours. It's $1/hr but billed per minute.

**Comment by abhagupta** [comment]
you are right @bootstraponline I wasn't clear on that. I will look into it in about an hour.

**Comment by abhagupta** [comment]
@bootstraponline  can you please check? thanks

**Comment by bootstraponline** [comment]
sure

File: Flank/src/main/java/com/walmart/otto/configurator/Configurator.java [hunk 2]
```diff
@@ -132,4 +132,4 @@ public boolean isFetchXMLFiles() {
     public void setFetchXMLFiles(boolean fetchXMLFiles) {
         this.fetchXMLFiles = fetchXMLFiles;
     }
-}
+}
```
Comment (bootstraponline) [inline Flank/src/main/java/com/walmart/otto/configurator/Configurator.java]: "No newline at end of file" is showing up in a few places.

**Comment by bootstraponline** [comment]
```
Operation completed over 112 objects/2.1 MiB.
$ gsutil --quiet rm -r gs://delta-essence-114723-flank/2017-06-02_00:40:33.176000_CndM/app.apk
$ gsutil --quiet rm -r gs://delta-essence-114723-flank/2017-06-02_00:40:33.176000_CndM/test.apk

Billable time:  112.0 min(s)

Estimated cost:  $2.24(virtual) $8.96(physical)

Combined test execution time: ~41 minutes

End time: 2017-06-01_08:51
```

Billable time looks right (112 VMs @ 1 minute each). The dollar amounts seem wrong though. The virtual device cost should be $1.86 (1/60.0 * 112) and physical cost $9.33 (5/60.0 * 112).

File: Flank/src/test/java/com/walmart/otto/reporter/PriceReporterTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    public void testGetTotalBillableTime(){
+        ArrayList<Integer> input = new ArrayList<Integer>();
+        input.add(30);
+        input.add(50);
+        input.add(73);
+        input.add(9);
+        Double totalBillableTime = PriceReporter.getTotalBillableTime(input);
+        assertEquals(totalBillableTime.toString(), "5.0");
+    }
+}
```
Comment (bootstraponline) [inline Flank/src/test/java/com/walmart/otto/reporter/PriceReporterTest.java]: missing newline

File: Flank/src/main/java/com/walmart/otto/reporter/TimeReporter.java [hunk 4]
```diff
@@ -28,4 +28,8 @@ public static String getEndTime() {
         Date date = new Date();
         return simpleDateFormat.format(date);
     }
+
+    public static List<Integer> getExecutionTimes(){
+        return executionTimes;
+    }
 }
```
Comment (bootstraponline) [inline Flank/src/main/java/com/walmart/otto/reporter/TimeReporter.java]: missing newline

**Comment by abhagupta** [comment]
hm...looks like its rounding before, adding 1 cent every time, becoming very big difference with large set of tests. Will look into it.

File: Flank/src/main/java/com/walmart/otto/reporter/TimeReporter.java (hunk 4 above)
Comment (abhagupta) [inline Flank/src/main/java/com/walmart/otto/reporter/TimeReporter.java]: is there a linting package in java? I work with javascript mostly, so not very sure. Hate to fix these manually.

Comment (bootstraponline) [inline Flank/src/main/java/com/walmart/otto/reporter/TimeReporter.java]: I have an open issue: https://github.com/TestArmada/flank/issues/24

java has a robust linting ecosystem and the better ones auto correct issues automatically. I will probably suggest adopting the Google Java format:

- https://github.com/google/google-java-format

this is a single command that updates all the code. 😄

**Comment by abhagupta** [comment]
@bootstraponline  
Should be fixed now. Let me know when you get a chance.

**Comment by bootstraponline** [comment]
Looks good!

```

Billable time:  111.0 min(s)

Estimated cost:  $1.85(virtual) $9.25(physical)

Combined test execution time: ~41 minutes

End time: 2017-06-02_09:30
```

**Comment by renas** [comment]
LGTM

**Pull request merged**